x-pack/plugins/monitoring/server/lib/cluster/get_clusters_summary.ts

```diff
@@ -47,7 +47,8 @@ export function getClustersSummary(
     } = cluster;
 
     const license = cluster.license || cluster.elasticsearch?.cluster?.stats?.license;
-    const version = cluster.version || cluster.elasticsearch?.version;
+    const version =
+      cluster.version || (cluster.elasticsearch?.cluster?.stats?.nodes?.versions || [])[0];
```

Comment:
```suggestion
    const version =
      cluster.version || cluster.elasticsearch?.cluster?.stats?.state?.version;
```

If the Metricbeat module was correct, this should be like this instead.

Reply:
I have no clue about how it's supposed to be, but `cluster.elasticsearch?.cluster?.stats?.state?.version` doesn't look correct. I would expect that to be the monotonically increasing cluster state "generation" clock.

Reply:
Just to note, I think we put the "monotonically increasing cluster state "generation" clock" value into `elasticsearch.version` because the issue mentions the version "number" going up over time.

Reply:
nit: should we sort the versions to ensure we provide a stable experience during an upgrade ? ie if the versions are unordered the UI could alternate between showing 7.17 and 8.0 until the upgrade is complete